modules/HardwareDrivers/PowerMeters/DZG_GSH01/app_layer.hpp

```diff
@@ -0,0 +1,357 @@
+        return "OK";
+    case CommandResult::GENERAL_ERROR:
+        return "General Error";
+    case CommandResult::OUT_OF_RANGE:
+        return "Out of Range";
+    case CommandResult::SECURITY_ACCESS_DENIED:
+        return "Security Access Denied";
+    case CommandResult::REJECTED:
+        return "Rejected";
+    case CommandResult::FORMAT_INVALID:
+        return "Format Invalid";
+    case CommandResult::NOT_AVAILABLE:
+        return "Not Available";
+    case CommandResult::BUSY:
+        return "Busy";
+    case CommandResult::PUBLIC_KEY_MISSING:
+        return "Public Key Missing";
+    case CommandResult::PROTOCOL_ERROR:
+        return "Protocol Error";
+    case CommandResult::TIMEOUT:
+        return "Timeout";
+    case CommandResult::COMMUNICATION_FAILED:
+        return "Communication Failed";
+    case CommandResult::PENDING:
+        return "Pending";
+    }
+    throw std::out_of_range("No known string conversion for provided enum of type CommandResult");
+}
+
+enum class UserIdStatus : std::uint8_t {
+    USER_NOT_ASSIGNED = 0x00,
+    USER_ASSIGNED = 0x01
+};
+
+enum class UserIdType : std::uint8_t {
+    NONE = 0,          // not available
+    DENIED = 1,        // not retrievable (e.g. two-factor-auth)
+    UNDEFINED = 2,     // type unknown / other
+    ISO14443 = 10,     // UID of RFID card according to ISO14443 (4 or 7 bytes HEX)
+    ISO15693 = 11,     // UID of RFID card according to ISO15693 (8 bytes HEX)
+    EMAID = 20,        // Electro-Mobility-Account-ID according to ISO/IEC15118 (14 or 15 bytes string)
+    EVCCID = 21,       // ID of an EV according to ISO/IEC15118 (max 6 bytes)
+    EVCOID = 30,       // EV-Contract-ID according to DIN91286
+    ISO7812 = 40,      // Identification-Card-Format according to ISO/IEC7812 (credit-/banking-cards, etc.)
+    CAR_TXN_NR = 50,   // Card-Transaction-Number (CardTxNbr) for credit- or banking-cards
+    CENTRAL = 60,      // centrally generated ID (no fixed format, e.g. UUID); OCPP 2.0
+    CENTRAL_1 = 61,    // centrally generated ID (no fixed format, e.g. start-via-SMS); (up to) OCPP 1.6
+    CENTRAL_2 = 62,    // centrally generated ID (no fixed format, e.g. start-by-operator); (up to) OCPP 1.6
+    LOCAL = 70,        // locally generated ID (no fixed format, e.g. UUID); OCPP 2.0
+    LOCAL_1 = 71,      // locally generated ID (no fixed format, e.g. generated by chargepoint); (up to) OCPP 1.6
+    LOCAL_2 = 72,      // locally generated ID (no fixed format, other); (up to) OCPP 1.6
+    PHONE_NUMBER = 80, // international phone number (leading '+' with country code)
+    KEY_CODE = 90      // private user key (no fixed format)
+};
+
+enum class ErrorCategory : std::uint8_t {
+    LAST = 0,
+    LAST_CRITICAL = 1,
+    FIRST = 2,
+    FIRST_CRITICAL = 3
+};
+
+enum class ErrorSource : std::uint8_t {
+    SYSTEM = 0,
+    COMMUNICATION = 1
+};
+
+enum class ApplicationBoardMode : std::uint8_t {
+    APPLICATION = 0,
+    ASSEMBLY = 1
+    // PRODUCTION = 2
```

Comment:
This looks similar to the missing enum value in AST_DC650 PR #1329 is this not needed?